src/rpc/core_rpc_server.cpp

```diff
@@ -2117,6 +2127,7 @@ namespace cryptonote
       return r;
 
     CHECK_PAYMENT_MIN1(req, res, COST_PER_BLOCK_HEADER, false);
+    db_rtxn_guard rtxn_guard(&m_core.get_blockchain_storage().get_db());
```

Comment:
This could be moved a bit further down, about 8 lines, to after the test about restriction.